File: PlacementCellBackend/Controllers/CRUD/CollageController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.Models;
using PlacementCellBackend.Services.CRUD.Interfaces;

namespace PlacementCellBackend.Controllers.CRUD;

[ApiController]
[Route("api/[controller]")]
public class CollageController : ControllerBase
{
    private readonly ICollageService _collageService;

    public CollageController(ICollageService collageService)
    {
        _collageService = collageService;
    }

    [HttpGet]
    public async Task<ActionResult<IEnumerable<College>>> GetAllColleges()
    {
        var colleges = await _collageService.GetAllCollegesAsync();
        return Ok(colleges);
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<College>> GetCollegeById(int id)
    {
        var college = await _collageService.GetCollegeByIdAsync(id);
        if (college == null)
            return NotFound();
        return Ok(college);
    }
    
    
    [HttpPost]
    public async Task<ActionResult<College>> CreateCollege(College college)
    {
        var created = await _collageService.CreateCollegeAsync(college);
        return CreatedAtAction(nameof(GetCollegeById), new { id = created.Id }, created);
    }
    
    
    
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/CRUD/CollageController.java
package com.example.placementicsbackend.controllers.CRUD;

import com.example.placementicsbackend.models.College;
import com.example.placementicsbackend.services.CRUD.Interfaces.ICollageService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/college")
@RequiredArgsConstructor
public class CollageController {

    private final ICollageService collageService;

    @GetMapping
    public ResponseEntity<List<College>> getAllColleges() {

        List<College> colleges = collageService.getAllColleges();

        return ResponseEntity.ok(colleges);
    }

    @GetMapping("/{id}")
    public ResponseEntity<College> getCollegeById(
            @PathVariable Integer id) {

        College college = collageService.getCollegeById(id);

        if (college == null) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.ok(college);
    }

    @PostMapping
    public ResponseEntity<College> createCollege(
            @RequestBody College college) {

        College created = collageService.createCollege(college);

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(created);
    }
}